Content.Shared/Strip/SharedStrippableSystem.cs

```diff
@@ -0,0 +1,51 @@
+using Content.Shared.DragDrop;
+using Content.Shared.Hands.Components;
+using Content.Shared.Strip.Components;
+
+namespace Content.Shared.Strip;
+
+public abstract class SharedStrippableSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<StrippingComponent, CanDropOnEvent>(OnCanDropOn);
+        SubscribeLocalEvent<StrippableComponent, CanDropEvent>(OnCanDrop);
+        SubscribeLocalEvent<StrippableComponent, DragDropEvent>(OnDragDrop);
+    }
+
+    private void OnDragDrop(EntityUid uid, StrippableComponent component, ref DragDropEvent args)
+    {
+        // If the user drags a strippable thing onto themselves.
+        if (args.Handled || args.Target != args.User)
+            return;
+
+        StartOpeningStripper(args.User, component);
+        args.Handled = true;
+    }
+
+    public virtual void StartOpeningStripper(EntityUid user, StrippableComponent component, bool openInCombat = false)
+    {
+
+    }
+
+    private void OnCanDropOn(EntityUid uid, StrippingComponent component, ref CanDropOnEvent args)
+    {
+        args.Handled = true;
+        args.CanDrop |= uid != args.Dragged &&
```

Comment:
AFAIK, `uid != args.Dragged` should always be true as the target cannot be the dragged entity? For the same reason the check in the other method (`args.Target != uid`) should also always be true.

Reply:
Also why do these two functions handle setting args.Handled differently? If there is some reason it needs to be done like that that should be changed because its not clear & will probably lead to bugs